pkg/cloud/services/network/vpc_test.go

```diff
@@ -170,82 +266,227 @@ func TestReconcileVPC(t *testing.T) {
+			},
+		},
+		{
+			name:    "Should return error if describe vpc returns empty list",
+			input:   &infrav1.VPCSpec{ID: "managed-vpc-exists", AvailabilityZoneUsageLimit: &usageLimit, AvailabilityZoneSelection: &selection},
+			wantErr: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(&ec2.DescribeVpcsOutput{
+					Vpcs: []*ec2.Vpc{},
+				}, nil)
+			},
+		},
+		{
+			name:    "Should return error if describe vpc returns more than 1 vpcs",
+			input:   &infrav1.VPCSpec{ID: "managed-vpc-exists", AvailabilityZoneUsageLimit: &usageLimit, AvailabilityZoneSelection: &selection},
+			wantErr: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(&ec2.DescribeVpcsOutput{
+					Vpcs: []*ec2.Vpc{
+						{
+							VpcId: aws.String("vpc_1"),
+						},
+						{
+							VpcId: aws.String("vpc_2"),
+						},
+					},
+				}, nil)
+			},
+		},
+		{
+			name:    "Should return error if vpc state is not available/pending",
+			input:   &infrav1.VPCSpec{ID: "managed-vpc-exists", AvailabilityZoneUsageLimit: &usageLimit, AvailabilityZoneSelection: &selection},
+			wantErr: true,
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeVpcs(gomock.AssignableToTypeOf(&ec2.DescribeVpcsInput{})).Return(&ec2.DescribeVpcsOutput{
+					Vpcs: []*ec2.Vpc{
+						{
+							VpcId: aws.String("vpc"),
+							State: aws.String("deleting"),
+						},
+					},
+				}, nil)
 			},
 		},
 	}
-
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			clusterScope, err := getClusterScope(tc.input)
+			g.Expect(err).NotTo(HaveOccurred())
 			ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
-
-			scheme := runtime.NewScheme()
-			_ = infrav1.AddToScheme(scheme)
-			awsCluster := &infrav1.AWSCluster{
-				ObjectMeta: metav1.ObjectMeta{Name: "test"},
-				Spec: infrav1.AWSClusterSpec{
-					NetworkSpec: infrav1.NetworkSpec{
-						VPC: *tc.input,
-					},
-				},
-			}
-			client := fake.NewClientBuilder().WithScheme(scheme).Build()
-			ctx := context.TODO()
-			client.Create(ctx, awsCluster)
-
-			clusterScope, err := scope.NewClusterScope(scope.ClusterScopeParams{
-				Cluster: &clusterv1.Cluster{
-					ObjectMeta: metav1.ObjectMeta{Name: "test-cluster"},
-				},
-				AWSCluster: awsCluster,
-				Client:     client,
-			})
-			if err != nil {
-				t.Fatalf("Failed to create test context: %v", err)
-			}
-
 			tc.expect(ec2Mock.EXPECT())
-
 			s := NewService(clusterScope)
 			s.EC2Client = ec2Mock
-			g := NewWithT(t)
 
 			err = s.reconcileVPC()
-			if tc.expectError {
+			if tc.wantErr {
 				g.Expect(err).ToNot(BeNil())
 				return
 			} else {
 				g.Expect(err).To(BeNil())
 			}
+			if !reflect.DeepEqual(tc.want, &clusterScope.AWSCluster.Spec.NetworkSpec.VPC) {
```

Comment:
Since we are using gomega across the files and tests, can we make this comparison using it, like below?
```
g.Expect(tc.want).To(Equal(&clusterScope.AWSCluster.Spec.NetworkSpec.VPC))
```